smqtk_descriptors/impls/descriptor_generator/image_descriptor_generator_wrapper.py

```diff
@@ -0,0 +1,110 @@
+from typing import Dict, Any, Iterable, TypeVar, Type, Set, Optional
+import numpy as np
+
+from smqtk_image_io import ImageReader
+from smqtk_core.configuration import (
+    from_config_dict,
+    make_default_config,
+    to_config_dict,
+)
+from smqtk_dataprovider import DataElement
+from smqtk_descriptors.utils.parallel import parallel_map
+from smqtk_descriptors.interfaces.descriptor_generator import \
+    DescriptorGenerator
+from smqtk_descriptors.interfaces.image_descriptor_generator import \
+    ImageDescriptorGenerator
+
+T = TypeVar("T", bound="ImageDescriptorGeneratorWrapper")
+
+
+class ImageDescriptorGeneratorWrapper (DescriptorGenerator):
+    """
+    Wrapper around some image descriptor generator, mediating image reading
+    into a numpy array.
+
+    :param smqtk_image_io.ImageReader image_reader:
+        ImageReader algorithm instance to facilitate reading image bytes
+        and producing pixel matrices.
+    :param smqtk_descriptors.interfaces.descriptor_generator_image.
+    ImageDescriptorGenerator image_descriptor_generator:
+        Image descriptor generator instance to perform descriptor generation
+        on loaded image matrices.
+    :param None|int image_load_threads:
+        Optional integer number of threads to parallelize image loading.
+        Thread parallelism only occurs if this is a positive integer
+        greater than 1.
+    """
+
+    @classmethod
+    def is_usable(cls) -> bool:
```

Comment:
meaning to remove this function entirely from `image_descriptor_generator_wrapper` ?

Reply:
Yes. Anywhere `is_usable` is implemented with a trivial `return True` implementation may be removed as an update upstream to `Pluggable` provides a default implementation of this method that does as much. The only reason to implement this method now is if the implementation is associated with optional availability logic, e.g. optional deps, hardware, etc., that would cause it to be "unavailable" during runtime.